tests/Feature/Controllers/CorsoDiStudiControllerTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Tests\Feature\Controllers;
+
+use Tests\TestCase;
+use App\Models\CorsoDiStudi;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class CorsoDiStudiControllerTest extends TestCase
+{
+    use RefreshDatabase; 
+
+    /** @test */
+    public function can_return_corsi_di_studi_as_a_json(): void
+    {
+        $this->seed(\DipartimentoTableSeeder::class);
+        $this->seed(\CorsoDiStudiTableSeeder::class);
+
+        $response = $this->json(
+            'GET',
+            '/api/v2/cds', 
+            ['anno_accademico' => '2017/2018']
+        );
+
+        $response->assertStatus(200);
+        $response->assertJson(CorsoDiStudi::where('anno_accademico', '2017/2018')->get()->toArray()); 
+    }
+
```

Comment:
Questi test mi sembrano buoni, pero', quando fai un test devi anche testare il caso in cui la route riceve cose "sbagliate", tipo `'anno_accademico' => ''`, oppure altro e ti devi assicurare che il risultato sia un json/array vuoto.

Devi essere duro, peggio di un prof. che sta esaminando uno studente :grin:

Reply:
(per il resto mi sembra tutto ok :wink: )

Reply:
Hai ragione, prossima PR aggiungo test per input malevoli!